scripts/launch_cluster.py

```diff
@@ -0,0 +1,642 @@
+                    },
+                    {
+                        "ParameterKey": "NodeInstanceType",
+                        "ParameterValue": instance_type
+                    },
+                    {
+                        "ParameterKey": "KeyName",
+                        "ParameterValue": keypair_name
+                    },
+                    {
+                        "ParameterKey": "VpcId",
+                        "ParameterValue": vpc_id
+                    },
+                    {
+                        "ParameterKey": "Subnets",
+                        "ParameterValue": ",".join(vpc_subnet_ids)
+                    }
+                ],
+                TimeoutInMinutes=15,
+                OnFailure='DELETE'
+            )
+
+            if response == None:
+                click.echo("Response is None, create worker group stack failed.")
+                sys.exit(1)
+            if not "StackId" in response:
+                click.echo("StackId not in response, create worker group stack failed.")
+                sys.exit(1)
+            click.echo("Initiated workers (ETA 5-20 mins)...")
+            click.echo("Waiting for workers stack creation to complete...")
+
+            try:
+                # This is a water which waits for the stack deployment to complete.
+                waiter = self._cf.get_waiter('stack_create_complete')
+                res = waiter.wait(
+                    StackName=workers_name
+                )
+            except Exception as e:
+                click.echo("Error: %s, gave up waiting for stack to create." % str(e))
+                sys.exit(1)
+            click.echo("Worker stack created.")
+
+        stack = self._cf.describe_stacks(StackName=workers_name)
+        node_instance_role=None
+        # We need NodeInstanceRole output.
+        for v in stack["Stacks"][0]["Outputs"]:
+            if v["OutputKey"] == "NodeInstanceRole": node_instance_role=v["OutputValue"]
+
+        click.echo("Node instance role: %s" % node_instance_role)
+        click.echo("*** Update worker auth.")
+
+        config = "apiVersion: v1\nkind: ConfigMap\nmetadata:\n  name: aws-auth\n" + \
+                "  namespace: kube-system\ndata:\n  mapRoles: |\n" + \
+                "    - rolearn: " + node_instance_role + "\n" + \
+                "      username: system:node:{{EC2PrivateDNSName}}\n" + \
+                "      groups:\n        - system:bootstrappers\n        - system:nodes\n"
+
+        click.echo("Write config map...")
+        worker_auth_file =  os.path.dirname(self._output_path) + "/aws-auth-cm.yaml"
+        with open(worker_auth_file, "w") as f:
+            f.write(config)
+        
+        # add the worker resource to cluster
+        resp = subprocess.call(["kubectl", "--kubeconfig=%s" % self._output_path, "apply",
+                                "-f", worker_auth_file])
+        if resp != 0:
+            click.echo("The kubectl command didn't work.")
```

Comment:
consider use `[subprocess.run](https://docs.python.org/3/library/subprocess.html#subprocess.run)`? And save the output and errors by `stdout=subprocess.PIPE, stderr=subprocess.PIPE`, so if exception occurs you can get the error messages in exception. As mentioned in doc:
> If check is true, and the process exits with a non-zero exit code, a CalledProcessError exception will be raised. Attributes of that exception hold the arguments, the exit code, and stdout and stderr if they were captured.